homeassistant/components/miele/light.py

```diff
@@ -0,0 +1,138 @@
+    """Class for defining light entities."""
+
+    types: tuple[MieleAppliance, ...]
+    description: MieleLightDescription
+
+
+LIGHT_TYPES: Final[tuple[MieleLightDefinition, ...]] = (
+    MieleLightDefinition(
+        types=(
+            MieleAppliance.OVEN,
+            MieleAppliance.OVEN_MICROWAVE,
+            MieleAppliance.STEAM_OVEN,
+            MieleAppliance.MICROWAVE,
+            MieleAppliance.COFFEE_SYSTEM,
+            MieleAppliance.HOOD,
+            MieleAppliance.STEAM_OVEN_COMBI,
+            MieleAppliance.WINE_CABINET,
+            MieleAppliance.WINE_CONDITIONING_UNIT,
+            MieleAppliance.WINE_STORAGE_CONDITIONING_UNIT,
+            MieleAppliance.STEAM_OVEN_MICRO,
+            MieleAppliance.WINE_CABINET_FREEZER,
+            MieleAppliance.STEAM_OVEN_MK2,
+        ),
+        description=MieleLightDescription(
+            key="light",
+            value_fn=lambda value: value.state_light,
+            translation_key="light",
+        ),
+    ),
+    MieleLightDefinition(
+        types=(MieleAppliance.HOOD,),
+        description=MieleLightDescription(
+            key="ambient_light",
+            value_fn=lambda value: value.state_ambient_light,
+            translation_key="ambient_light",
+        ),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: MieleConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the light platform."""
+    coordinator = config_entry.runtime_data
+
+    async_add_entities(
+        MieleLight(coordinator, device_id, definition.description)
+        for device_id, device in coordinator.data.devices.items()
+        for definition in LIGHT_TYPES
+        if device.device_type in definition.types
+    )
+
+
+class MieleLight(MieleEntity, LightEntity):
+    """Representation of a Light."""
+
+    entity_description: MieleLightDescription
+    _attr_color_mode = ColorMode.ONOFF
+    _attr_supported_color_modes = {ColorMode.ONOFF}
+
+    def __init__(
+        self,
+        coordinator: MieleDataUpdateCoordinator,
+        device_id: str,
+        description: MieleLightDescription,
+    ) -> None:
+        """Initialize the light."""
+        super().__init__(coordinator, device_id, description)
+        self.api = coordinator.api
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return current on/off state."""
+        return self.entity_description.value_fn(self.device) == LIGHT_ON
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the light."""
+        await self.async_turn_light(LIGHT_ON)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the light off."""
+        await self.async_turn_light(LIGHT_OFF)
+
+    async def async_turn_light(self, mode: int) -> None:
+        """Set light to mode."""
+        light_type = (
+            AMBIENT_LIGHT if self.entity_description.key == "ambient_light" else LIGHT
+        )
```

Comment:
I would make this a separate field of the description, this way the class stays truly generic